geomstats/geometry/siegel.py

```diff
@@ -0,0 +1,614 @@
+        aux_6 = identity - aux_5
+        factor_3 = gs.linalg.inv(aux_6)
+        prod_1 = gs.einsum("...ij,...jk->...ik", factor_1, factor_2)
+        prod_2 = gs.einsum("...ij,...jk->...ik", prod_1, factor_3)
+        point_image = gs.einsum("...ij,...jk->...ik", prod_2, factor_4)
+        return point_image
+
+    def exp(self, tangent_vec, base_point):
+        """Compute the Siegel exponential map.
+
+        Compute the exponential map at base_point of tangent_vec.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., n, n]
+            Point on the manifold.
+
+        Returns
+        -------
+        exp : array-like, shape=[..., n, n]
+            Point on the manifold.
+        """
+        tangent_vec_at_zero = self.tangent_vec_from_base_point_to_zero(
+            tangent_vec=tangent_vec, base_point=base_point
+        )
+
+        exp_zero = self.exp_at_zero(tangent_vec_at_zero)
+
+        exp = self.isometry(point=exp_zero, point_to_zero=-base_point)
+
+        return exp
+
+    @staticmethod
+    def log_at_zero(point):
+        """Compute the Siegel logarithm map at zero.
+
+        Compute the logarithm map at zero (null matrix) of point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+
+        Returns
+        -------
+        log_at_zero : array-like, shape=[..., n, n]
+            Riemannian logarithm at zero (null matrix).
+        """
+        data_type = point.dtype
+        identity = create_identity_mat(point.shape, dtype=data_type)
+        point_transconj = ComplexMatrices.transconjugate(point)
+        aux_1 = gs.einsum("...ij,...jk->...ik", point, point_transconj)
+        aux_2 = HermitianMatrices.powerm(aux_1, 1 / 2)
+        num = identity + aux_2
+        den = identity - aux_2
+        inv_den = HermitianMatrices.powerm(den, -1)
+        frac = gs.einsum("...ij,...jk->...ik", num, inv_den)
+        factor_1 = gs.linalg.logm(frac)
+        factor_2 = HermitianMatrices.powerm(aux_2, -1)
+        prod_1 = gs.einsum("...ij,...jk->...ik", factor_1, factor_2)
+        log_at_zero = gs.einsum("...ij,...jk->...ik", prod_1, point)
+        log_at_zero *= 0.5
+        return log_at_zero
+
+    @staticmethod
+    def tangent_vec_from_zero_to_base_point(tangent_vec, base_point):
+        """Transport a tangent vector from zero to a base point.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., n, n]
+            Tangent vector at zero.
+        base_point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+
+        Returns
+        -------
+        tangent_vec_at_base_point : array-like, shape=[..., n, n]
+            Tangent vector at the base point.
+        """
+        data_type = (tangent_vec + base_point).dtype
```

Comment:
use `base_point`. no need to create variable.

Reply:
Done.